api/src/modules/admin-regions/admin-region.repository.ts

```diff
@@ -60,17 +65,27 @@ export class AdminRegionRepository extends ExtendedTreeRepository<
           st_setsrid($1::geometry, 4326),
           st_setsrid(g."theGeom"::geometry, 4326)
           )
-        AND ST_DWithin(g."theGeom" , st_setsrid($1::geometry, 4326), 1000)
-        AND a.level IS NOT NULL
         ORDER BY a.level DESC
-        LIMIT 1`,
+        LIMIT 3`,
```

Comment:
shouldn't the "order by a.level" make this and the "reduce()" dedundant? can you give me more detail as to what exactly happens to make this necesary?

Reply:
There are 2 functions to determine de admin-region given some coordinates. In the use-case of this method we can't determine the granularity of the intersection, it might end-up returning a intersection which it's level might be null (or not)

That's why I decided to retrieve the final value within the API and make sure PostGIS retrieves some intersections.
Which would be a better approach?